spec/asyncapi.md

```diff
@@ -998,9 +998,9 @@ Describes a message received on a given channel and operation.
 Field Name | Type | Description
 ---|:---:|---
 <a name="messageObjectHeaders"></a>headers | [Schema Object](#schemaObject) &#124; [Reference Object](#referenceObject) | Schema definition of the application headers. Schema MUST be of type "object". It **MUST NOT** define the protocol headers.
-<a name="messageObjectPayload"></a>payload | `any` | Definition of the message payload. It can be of any type but defaults to [Schema object](#schemaObject).
+<a name="messageObjectPayload"></a>payload | `any` | Definition of the message payload. It can be of any type but defaults to [Schema object](#schemaObject). It must match the schema format, including encoding type - e.g Avro should be inlined as either a YAML or JSON object NOT a string to be parsed as YAML or JSON.
```

Comment:
One suggestion regarding formal language:

```suggestion
<a name="messageObjectPayload"></a>payload | `any` | Definition of the message payload. It can be of any type but defaults to [Schema object](#schemaObject). It MUST match the schema format defined in [`schemaFormat`](#messageObjectSchemaFormat), including encoding type - e.g Avro should be inlined as either a YAML or JSON object NOT a string to be parsed as YAML or JSON.
```

And why can't it be inlined as a string? I think it's fine to have a YAML Avro document inlined in a JSON AsyncAPI file:

```json
{
  "payload": "type: record\nfields:\n  - name: displayName\n    type: string"
}
```

I mean, not the best UX but may be fine for tooling-generated AsyncAPI files that are not meant to be consumed by humans.

Reply:
> And why can't it be inlined as a string? I think it's fine to have a YAML Avro document inlined in a JSON AsyncAPI file

This was the reason for raising the PR - the JS parser treats a `String` as a primitive Avro schema (the string is the name of a primitive type e.g `"string"`). And as such, the inlined schema is not able to be parsed.

If however the thought is that it should be acceptable, I'm happy to raise an issue instead on the parser to support it

Reply:
Oh yeah! You're completely right. A string is a valid Avro schema and is a valid JSON Schema too. Thanks for noticing!